server/src/test/java/org/opensearch/extensions/ExtensionsManagerWithIdentityTests.java

```diff
@@ -0,0 +1,231 @@
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.common.util.PageCacheRecycler;
+import org.opensearch.identity.IdentityService;
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.noop.NoopSubject;
+import org.opensearch.indices.breaker.NoneCircuitBreakerService;
+import org.opensearch.plugins.IdentityPlugin;
+import org.opensearch.rest.RestController;
+import org.opensearch.test.FeatureFlagSetter;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.client.NoOpNodeClient;
+import org.opensearch.test.transport.MockTransportService;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.nio.MockNioTransport;
+import org.opensearch.usage.UsageService;
+
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.opensearch.test.ClusterServiceUtils.createClusterService;
+
+public class ExtensionsManagerWithIdentityTests extends OpenSearchTestCase {
+
+    private FeatureFlagSetter featureFlagSetter;
+    private TransportService transportService;
+    private ActionModule actionModule;
+    private RestController restController;
+    private SettingsModule settingsModule;
+    private ClusterService clusterService;
+    private IdentityService identityService;
+
+    private Setting customSetting = Setting.simpleString("custom_extension_setting", "none", Property.ExtensionScope);
+    private NodeClient client;
+    private MockNioTransport transport;
+    private Path extensionDir;
+    private final ThreadPool threadPool = new TestThreadPool(ExtensionsManagerWithIdentityTests.class.getSimpleName());
+    private final List<String> extensionsYmlLines = Arrays.asList(
+        "extensions:",
+        "   - name: firstExtension",
+        "     uniqueId: uniqueid1",
+        "     hostAddress: '127.0.0.0'",
+        "     port: '9300'",
+        "     version: '0.0.7'",
+        "     opensearchVersion: '3.0.0'",
+        "     minimumCompatibleVersion: '3.0.0'",
+        "     custom_extension_setting: 'custom_setting'",
+        "   - name: secondExtension",
+        "     uniqueId: 'uniqueid2'",
+        "     hostAddress: '127.0.0.1'",
+        "     port: '9301'",
+        "     version: '3.14.16'",
+        "     opensearchVersion: '2.0.0'",
+        "     minimumCompatibleVersion: '2.0.0'"
```

Comment:
What do you think about trying one where the version is not up to the minimum required or some other mis-configured options?

Reply:
I'd like to keep the tests introduced in this PR related to the change that this PR introduces.